pkg/auth/tokenparser_test.go

```diff
@@ -0,0 +1,111 @@
+package auth_test
+
+import (
+	"log"
+	"os"
+	"testing"
+
+	"github.com/codeready-toolchain/registration-service/pkg/auth"
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	testutils "github.com/codeready-toolchain/registration-service/test"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTokenParser(t *testing.T) {
+	// create logger and registry.
+	logger := log.New(os.Stderr, "", 0)
+	configRegistry := configuration.CreateEmptyRegistry()
+
+	// create test keys
+	tokengenerator := testutils.NewTokenManager()
+	kid0 := uuid.NewV4().String()
+	_, err := tokengenerator.AddPrivateKey(kid0)
+	require.NoError(t, err)
+	kid1 := uuid.NewV4().String()
+	_, err = tokengenerator.AddPrivateKey(kid1)
+	require.NoError(t, err)
+
+	// create two test tokens, both valid
+	username0 := uuid.NewV4().String()
+	identity0 := &testutils.Identity{
+		ID:       uuid.NewV4(),
+		Username: username0,
+	}
+	email0 := identity0.Username + "@email.tld"
+	jwt0, err := tokengenerator.GenerateSignedToken(*identity0, kid0, testutils.WithEmailClaim(email0))
+	require.NoError(t, err)
+	username1 := uuid.NewV4().String()
+	identity1 := &testutils.Identity{
+		ID:       uuid.NewV4(),
+		Username: username1,
+	}
+	email1 := identity1.Username + "@email.tld"
+	jwt1, err := tokengenerator.GenerateSignedToken(*identity1, kid1, testutils.WithEmailClaim(email1))
+	require.NoError(t, err)
+
+	// create invalid test token (wrong set of claims, no email), signed with key1
+	username_invalid := uuid.NewV4().String()
+	identity_invalid := &testutils.Identity{
+		ID:       uuid.NewV4(),
+		Username: username_invalid,
+	}
+	jwt_invalid, err := tokengenerator.GenerateSignedToken(*identity_invalid, kid1)
```

Comment:
you could move this part to the only subtest that uses it, to simplify the understanding of all this setup

Reply:
Done d7a998c.